README.md
# Flutter Kpay Kit

KBZ Mobile Payment package for Flutter

## Getting Started



## Installation
```dart
dependencies:
  flutter_kpay_kit:latest version
```


## Usage


### Android


[UAT version](https://drive.google.com/drive/folders/1y8rwhg8tF35U_S5CKbROrR-FXI8B-0S7?usp=share_link)   
- add or overwirte uat kbzsdk.arr under yourproject/android/app/lib

[Product version](https://drive.google.com/drive/folders/1YFiXmBNv1zAM4kZb32A8ujrMlG08pxd4?usp=share_link)  
- add or overwirte prod kbzsdk.arr under yourproject/android/app/lib



CallBack Function
- add following code in android/app/src/main/manifest.xml

```xml
<manifest xmlns:android="http://schemas.android.com/apk/res/android"
....
	<activity android:name="com.kbzbank.payment.sdk.callback.CallbackResultActivity" android:theme="@android:style/Theme.NoDisplay" android:exported="true"
....
```
### IOS
App project configuration in the Info. Add kbzpay pist white list
ios/Runner/Info.plist
```plist
<key>LSApplicationQueriesSchemes</key>
	<array>
		<string>kbzpay</string>
	</array>
```

### Payment callback
Payment callback, payment completion or payment cancellation, currently there are only two states. The callback parameter is returned as an OpenUrl, as shown below


1：Pay for success，
2：Payment failed,
3: Payment cancel

the remaining fields are reserved for later addition。

## Example
```dart
import 'package:flutter_kpay_kit/flutter_kpay_kit.dart';

  void startPay() {
    FlutterKpayKit.startPay(
            merchCode: this.merchCode,
            // "10000",
            appId: this.appId,
            //"kp4c1706c8675a45fghjklrskyf",
            signKey: this.signKey,
            //"123",
            orderId: this.orderId,
            //"294",
            amount: this.amount,
            title: "title",
            urlScheme: "",
            //Only Ios
            isProduction: false,
            notifyURL: 'http://test.payment.com/notify')
        .then((res) {
      print('startPay' + res.toString());
    });
  }

void openKpay() {
  FlutterKpayKit.connectKBZPay(
      prepayID: '123',
      merchCode: '123',
      appId: '123',
      //Only Ios
      urlScheme: 'https',
      signKey: 'signKey')
      .then((res) {
    Map response = json.decode(res);
    String result = response["Response"]["result"];
    if (result == "FAIL") {
      _messangerKey.currentState!.showSnackBar(
        SnackBar(content: Text(response["Response"]["msg"])),
      );
    }
  });
}

    
  @override
  void initState() {
    super.initState();
    FlutterKpayKit.onPayStatus().listen((data) {
      print('onPayStatus $data');
    });
  }
```

## Getting Started

This project is a starting point for a Flutter
[plug-in package](https://flutter.dev/developing-packages/),
a specialized package that includes platform-specific implementation code for
Android and/or iOS.

For help getting started with Flutter development, view the
[online documentation](https://flutter.dev/docs), which offers tutorials,
samples, guidance on mobile development, and a full API reference.



CHANGELOG.md
## **[0.0.7] - 22 DEC 23**
- fix iOS bug for Kpay payment.

## **[0.0.6] - 22 DEC 23**
- add iOS code for Kpay payment.

## **[0.0.5] - 21 DEC 23**
- New method (connectKBZPay) for open KBZPay mobile application.

## **[0.0.4] - 20 DEC 23**
- Update flutter version and package version.

## **[0.0.3] - 14 NOV 22**
- Update Readme and Changelog documentation
- Add response message for Payment Fail

## **[0.0.2] - 14 NOV 22**
- Fixed Payment callback error

## **[0.0.1] - 9 NOV 22**
- Initial release, forked from a primary project.

example/lib/main.dart
import 'dart:convert';

import 'package:flutter/material.dart';
import 'dart:async';

import 'package:flutter/services.dart';
import 'package:flutter_kpay_kit/flutter_kpay_kit.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatefulWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  State<MyApp> createState() => _MyAppState();
}

class _MyAppState extends State<MyApp> {
  String orderId = '123',
      merchCode = '10000',
      appId = 'kp6540b7940f3b44f8b275ec26988062',
      signKey = '123';
  double amount = 5000;

  GlobalKey _formKey = new GlobalKey<FormState>();

  final _messangerKey = GlobalKey<ScaffoldMessengerState>();

  @override
  void initState() {
    super.initState();
    FlutterKpayKit.onPayStatus().listen((data) {
      print('onPayStatus $data');
    });
  }

  void success(dynamic data) {
    print(data);
  }

  void error(dynamic data) {
    print(data);
  }

  void startPay() {
    FlutterKpayKit.startPayment(
            merchCode: merchCode,
            // "10000",
            appId: appId,
            //"kp4c1706c8675a45fghjklrskyf",
            signKey: signKey,
            //"123",
            orderId: orderId,
            //"294",
            amount: amount,
            title: "title",
            urlScheme: "",
            //Only Ios
            isProduction: false,
            notifyURL: 'http://test.payment.com/notify')
        .then((res) {
      Map response = json.decode(res);
      String result = response["Response"]["result"];
      if (result == "FAIL") {
        _messangerKey.currentState!.showSnackBar(
          SnackBar(content: Text(response["Response"]["msg"])),
        );
      }
    });
  }

  void openKpay() {
    FlutterKpayKit.connectKBZPay(
            prepayID: '123',
            merchCode: '123',
            appId: '123',
            //Only Ios
            urlScheme: 'https',
            signKey: 'signKey')
        .then((res) {
      Map response = json.decode(res);
      String result = response["Response"]["result"];
      if (result == "FAIL") {
        _messangerKey.currentState!.showSnackBar(
          SnackBar(content: Text(response["Response"]["msg"])),
        );
      }
    });
  }

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      scaffoldMessengerKey: _messangerKey,
      home: Scaffold(
        appBar: AppBar(
          title: const Text('Plugin example app'),
        ),
        body: Padding(
          padding: const EdgeInsets.symmetric(vertical: 16.0, horizontal: 24.0),
          child: Form(
            autovalidateMode: AutovalidateMode.always,
            key: _formKey,
            child: ListView(
              children: <Widget>[
                TextFormField(
                  initialValue: orderId,
                  decoration: const InputDecoration(
                    label: Text('Order ID:'),
                    border: UnderlineInputBorder(
                      borderSide: BorderSide(color: Colors.cyan),
                    ),
                  ),
                  validator: (String? value) => value == null ? '' : null,
                  onSaved: (String? value) => orderId = value!,
                ),
                const SizedBox(height: 15.0),
                TextFormField(
                  initialValue: merchCode,
                  decoration: const InputDecoration(
                    label: Text(
                      'Merch Code:',
                    ),
                    border: UnderlineInputBorder(
                      borderSide: BorderSide(color: Colors.cyan),
                    ),
                  ),
                  validator: (String? value) => value == null ? '' : null,
                  onSaved: (String? value) => merchCode = value!,
                ),
                const SizedBox(height: 15.0),
                TextFormField(
                  initialValue: appId,
                  decoration: const InputDecoration(
                    label: Text(
                      'App ID:',
                    ),
                    border: UnderlineInputBorder(
                      borderSide: BorderSide(color: Colors.cyan),
                    ),
                  ),
                  validator: (String? value) => value == null ? '' : null,
                  onSaved: (String? value) => appId = value!,
                ),
                const SizedBox(height: 15.0),
                TextFormField(
                  initialValue: signKey,
                  decoration: const InputDecoration(
                    label: Text('Sign Key'),
                    border: UnderlineInputBorder(
                      borderSide: BorderSide(color: Colors.cyan),
                    ),
                  ),
                  validator: (String? value) => value == null ? '' : null,
                  onSaved: (String? value) => signKey = value!,
                ),
                TextFormField(
                  initialValue: '10000',
                  keyboardType: TextInputType.number,
                  decoration: const InputDecoration(
                    label: Text('Amount:'),
                    border: UnderlineInputBorder(
                      borderSide: BorderSide(color: Colors.cyan),
                    ),
                  ),
                  validator: (String? value) => value == null ? '' : null,
                  onSaved: (String? value) => amount = double.parse(value!),
                ),
                const SizedBox(height: 15.0),
                ElevatedButton(
                    child: const Text('Pay'),
                    onPressed: () {
                      startPay();
                    }),
              ],
            ),
          ),
        ),
      ),
    );
  }
}
